src/gc/gc.h

```diff
@@ -352,19 +197,7 @@ void record_changed_seg (uint8_t* start, uint8_t* end,
 void record_global_mechanism (int mech_index);
 #endif //GC_CONFIG_DRIVEN
 
-//constants for the flags parameter to the gc call back
-
-#define GC_CALL_INTERIOR            0x1
-#define GC_CALL_PINNED              0x2
-#define GC_CALL_CHECK_APP_DOMAIN    0x4
-
-//flags for GCHeap::Alloc(...)
-#define GC_ALLOC_FINALIZE 0x1
-#define GC_ALLOC_CONTAINS_REF 0x2
-#define GC_ALLOC_ALIGN8_BIAS 0x4
-#define GC_ALLOC_ALIGN8 0x8
-
-class GCHeap {
+class GCHeap : public IGCHeap {
```

Comment:
The abstract ::GCHeap is pretty close to the interface definition that you are introducing. It is just not obvious from the name that it is really interface.

It may be better to do this in two steps:
- Rename ::GCHeap to IGCHeap. It is very boring change touching many places. Maybe better done as independent PR.
- Move a bunch of stuff around. The interesting stuff in this change that's worth reviewing.